harness/determined/cli/workspace.py

```diff
@@ -232,22 +244,30 @@ def create_workspace(args: argparse.Namespace) -> None:
         defaultComputePool=args.default_compute_pool,
         defaultAuxPool=args.default_aux_pool,
     )
-    if args.namespace:
-        # When args.cluster_name is unspecified, the string "null" assumes its value in the
-        # clusterNamespacePairs dictionary. To avoid that, we substitute the argument's value with
-        # the empty string if it's null.
-        cluster_name = args.cluster_name or ""
-        namespace_meta = bindings.v1WorkspaceNamespaceMeta(namespace=args.namespace)
+
+    cluster_name = args.cluster_name or ""
+    requested_namespace = args.namespace or ""
+    if args.namespace or args.auto_create_namespace:
+        # When args.cluster_name and/or args.namespace is unspecified, the string "null" assumes its
+        # value in the clusterNamespacePairs dictionary. To avoid that, we substitute the argument's
+        # value with the empty string if it's null.
+        namespace_meta = bindings.v1WorkspaceNamespaceMeta(
+            namespace=requested_namespace,
+            autoCreateNamespace=args.auto_create_namespace,
+            autoCreateAllNamespaces=args.auto_create_all_namespaces,
+        )
         content.clusterNamespaceMeta = {cluster_name: namespace_meta}
-    w = bindings.post_PostWorkspace(sess, body=content).workspace
+    resp = bindings.post_PostWorkspace(sess, body=content)
+    w = resp.workspace
 
     if args.json:
         render.print_json(w.to_json())
     else:
         render_workspaces([w])
 
-    if args.namespace:
-        print(f"Workspace {w.name} is bound to namespace {args.namespace}")
+    if len(resp.namespaceBindings) > 0:
```

Comment:
style: prefer `if rest.namespaceBindings`, since empty dictionaries [evaluate to False](https://docs.python.org/2/library/stdtypes.html#truth-value-testing).

Reply:
Good point, changed!